packages/amplify_datastore_plugin_interface/lib/src/types/temporal/temporal.dart

```diff
@@ -39,7 +39,7 @@ class Temporal {
         caseSensitive: false,
         multiLine: false);
 
-    var match = regExp.matchAsPrefix(offsetString);
+    var match = regExp.matchAsPrefix(offsetString)!;
```

Comment:
I feel this is an "escape hatch" or a work around using null safety. Null safety very correctly told us that the value here is nullable and your code expects a non-nullable and we are forcing (or overridding) the null safety here.

In general, I think we should avoid use of force un-wrapping (i.e. `!` operator) as much as we can and fail gracefully when we detect a null.

Reply:
Alright that makes sense, added a graceful case to return a blank duration if no match was found.  However, isn't the ! operator necessary when making any access on a nullable field? 

How would you write this code without using the ! operator? 

![image](https://user-images.githubusercontent.com/11478656/114105706-1a56d200-9882-11eb-9ab4-72f38df235b4.png)

Reply:
Do you think it would be better if we called match?.group everwhere we could?  And just propagated the nullable type as far down as possible?

Reply:
hmm I think this is one of those places where the language (compiler) is not "smart enough" to know that the var is guaranteed to be non-nullable and so force unwrap is fine. 

But for cases like below

```diff
   static AmplifyException fromMap(Map<String, String> serializedException) {
-    return AmplifyException(serializedException["message"],
+    return AmplifyException(serializedException["message"]!,
```

We know that the `serializedException["message"]` should be non-null, and so we should check and throw an exception if it is and in the else use it without the force unwrap (like swift `guard`)

Reply:
Sounds good updated that example and combed through the code looking for similar changes to make.